### ISSUE TITLE
Consider implementing signatures that take input in a way alike to compiletime-string-to-bytearray

### ISSUE THREAD

**Issue by cristeigabriel** [danielkrupinski/Osiris#2865]
https://github.com/danielkrupinski/Osiris/blob/0a0a63b9a89dfbad759d18ab916f84abcb8e845b/Osiris/Memory.cpp#L105

This isn't that big of an issue, but perhaps could be of help, it'd make no need to process the text itself. I suppose this can be done also like in projects alike.

The project [that I'm referring's](https://github.com/cristeigabriel/compiletime-string-to-bytearray) generic implementation allows for usage of IDA-style signatures, and is computed at compile-time into a constexpr std::array, and can be made compliant with the current type of signatures, by removing the implicit _char_to_int call, which is something I'll implement after opening this issue.

Regards.

**Comment by exploitmafia** [comment]
I don't think this would be worth the additional code and potential compile time for some users, when IDA directly supports creating code signatures that work in the way Osiris does.

**Comment by cristeigabriel** [comment]
> I don't think this would be worth the additional code and potential compile time for some users, when IDA directly supports creating code signatures that work in the way Osiris does.
> 
> Altough this is a cool thing and cool you can do this compile-time.

I don't understand your reply. What I've linked IS generating the byte arrays at compile-time, from HEX strings, and supports a skip wildcard, which to use IDA style signatures can be set to ? or C
